src/uct/ib/ud/accel/ud_mlx5.c

```diff
@@ -904,6 +904,8 @@ uct_ud_mlx5_query_tl_devices(uct_md_h md,
                                      tl_devices_p, num_tl_devices_p);
 }
 
-UCT_TL_DEFINE(&uct_ib_component, ud_mlx5, uct_ud_mlx5_query_tl_devices,
-              uct_ud_mlx5_iface_t, "UD_MLX5_", uct_ud_mlx5_iface_config_table,
-              uct_ud_mlx5_iface_config_t);
+UCT_TL_REGISTER_DEF(&uct_ib_component, ud_mlx5, uct_ud_mlx5_query_tl_devices,
+                    uct_ud_mlx5_iface_t, "UD_MLX5_",
+                    uct_ud_mlx5_iface_config_table, uct_ud_mlx5_iface_config_t);
+
+UCT_IB_TL(ud_mlx5)
```

Comment:
can it be a part of `UCT_TL_REGISTER_DEF` or another wrapper macro? So that there will be just one macro needed

Reply:
> UCT_TL_REGISTER_DEF

no: ```UCT_TL_REGISTER_DEF``` is general macro which is used to define constructors with custom actions to initialise

Reply:
ic, but what is about having additional macro like `UCT_IB_TL_REGISTER_DEF` which would also call `UCT_IB_TL`?